dotcom-rendering/src/web/lib/transparentColour.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Transforms a hex colour to a transparent one of the same colour.
+ *
+ * The alternative [#RRGGBBAA notation is not well supported][caniuse],
+ * so we used the more widely supported rgba() notation.
+ *
+ * [caniuse]: https://caniuse.com/css-rrggbbaa
+ *
+ * @param colour the hex string, e.g. #C70000
+ * @param opacity optional opacity 0-1. Defaults to 0.5 (50%)
+ * @returns an functional colour string, e.g. rgba(199, 0, 0, 0.5)
+ */
+export const transparentColour = (
+	colour: string,
+	opacity = 0.5,
+): `rgba(${number}, ${number}, ${number}, ${number})` => {
+	// extract hex colours as numbers
+	const r = parseInt(colour.slice(1, 3), 16);
```

Comment:
I really ❤️  this solution.
Just one thought that occurred to me... I think this might break if we pass in a 3 digit hex colour, for instance #000 or #FFF (which is valid hex notation). 
I don't think it's an issue though,  because the colours in the palette all have 6 digits. 
Maybe to be extra safe we could, at some point, add a [linting rule](https://stylelint.io/user-guide/rules/list/color-hex-length/) to only allow 6 digit hex colours?

Reply:
Tbh you're right, this could be safer.

I've updated the method to allow short hexes, and added tests.